modules/core/include/opencv2/core/utility.hpp

```diff
@@ -513,30 +513,36 @@ class CV_EXPORTS AutoLock
     AutoLock& operator = (const AutoLock&);
 };
 
+// TLS interface
 class CV_EXPORTS TLSDataContainer
 {
-private:
-    int key_;
 protected:
-    TLSDataContainer();
-    virtual ~TLSDataContainer();
-public:
-    virtual void* createDataInstance() const = 0;
-    virtual void deleteDataInstance(void* data) const = 0;
+    TLSDataContainer() { key = (size_t)-1; };
+    virtual ~TLSDataContainer() {};
 
+    void  getKey();
     void* getData() const;
+    void  releaseKey();
+
+private:
```

Comment:
Is this "OPENCV_ABI_COMPATIBILITY" supported by build system and will be defined automatically? It should or I will get another error.

If it is not, may be something like "OPENCV_ABI_300" would be more suitable? To understand which ABI version is used.

Also, what is the criteria for keeping or breaking ABI compatibility for a release?